feat: Add RTL Support to Widget

iamsivin opened the pull request:
# Pull Request Template

## Description

This PR adds RTL support to the web widget for improved right-to-left language compatibility, updates colors, and cleans up code.

Fixes https://linear.app/chatwoot/issue/CW-4089/rtl-issues-on-widget
https://github.com/chatwoot/chatwoot/issues/9791

Other PR: https://github.com/chatwoot/chatwoot/pull/11016

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?


| Dark | Light |
| ------------- | ------------- |
| <img width="434" alt="image" src="https://github.com/user-attachments/assets/f5128feb-0404-4dc7-8a9f-405adcd62fe5" />  | <img width="434" alt="image" src="https://github.com/user-attachments/assets/cb660de6-b0aa-4e8c-af8a-4349df0f2327" />  |
| <img width="434" alt="image" src="https://github.com/user-attachments/assets/6d845272-eecc-4862-8af8-9c366227576f" /> | <img width="434" alt="image" src="https://github.com/user-attachments/assets/1140b4f3-5361-4ea8-ae3e-2428b054af97" />   |
| <img width="434" alt="image" src="https://github.com/user-attachments/assets/ce1b27dc-71d6-4b73-abc7-7b6108b4cefa" />  | <img width="434" alt="image" src="https://github.com/user-attachments/assets/69fcf7b7-78e4-40de-9634-738b38d30c0b" />  |
| <img width="431" alt="image" src="https://github.com/user-attachments/assets/eb445e05-20c2-48aa-b9f4-c472eb04e73d" /> | <img width="431" alt="image" src="https://github.com/user-attachments/assets/a8beafca-5eee-4f50-8c14-6853fc96aa21" />  |
| <img width="431" alt="image" src="https://github.com/user-attachments/assets/6ed05640-5c42-4137-8e46-262da9fb956e" /> | <img width="431" alt="image" src="https://github.com/user-attachments/assets/067ae973-bc0b-4125-8e3f-e56621d38251" /> |
| <img width="431" alt="image" src="https://github.com/user-attachments/assets/c32e712e-8127-437c-86b3-0c310ff672cd" /> | <img width="431" alt="image" src="https://github.com/user-attachments/assets/a26a7c32-5ae8-4262-85c9-64141f27a08b" /> |
| <img width="440" alt="image" src="https://github.com/user-attachments/assets/a296df3e-f81f-4e4b-bb43-2f4c02d097e0" /> | <img width="440" alt="image" src="https://github.com/user-attachments/assets/4590cdb5-ab3a-4570-a857-09d1eb7ed77b" /> |
| <img width="431" alt="image" src="https://github.com/user-attachments/assets/709fad26-913f-41f2-a26e-2816d2573243" /> | <img width="431" alt="image" src="https://github.com/user-attachments/assets/38ac2058-b4c5-41f5-9783-85892552bff4" /> |



## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

muhsin-k commented:
@iamsivin Please check this changes are intentional (font size and spacing)?
| Before | After |
| ------------- | ------------- |
| <img width="434" alt="image" src="https://github.com/user-attachments/assets/4ccb40d4-bcb3-4846-8edf-82b3a78270f5" />  | <img width="434" alt="image" src="https://github.com/user-attachments/assets/48dc573a-e770-4295-91c7-05bbb3acfe12" />  |

muhsin-k commented:
Noted this issue. Of course we can handle this in a separate PR. We don't need to show this skeleton loader if the article has failed to load.

![CleanShot 2025-03-07 at 15 57 19@2x](https://github.com/user-attachments/assets/8469e090-bd38-43f1-be9a-41db99afb404)

iamsivin commented:
@muhsin-k 
> Please check this changes are intentional)?

It was not intentional, updated the PR https://github.com/chatwoot/chatwoot/pull/11022/commits/964ab9f694a64d2304c97dfb241ea4e1e3603c42.
> Noted this issue while testing the widget, of course we can handle this in a separate PR. We don't need to show this skeleton loader if the article has failed to load.

Fixed https://github.com/chatwoot/chatwoot/pull/11022/commits/964ab9f694a64d2304c97dfb241ea4e1e3603c42.

muhsin-k commented:
@iamsivin Check the gap between the button and last form element?
![CleanShot 2025-03-10 at 20 55 45@2x](https://github.com/user-attachments/assets/2c89a979-6ee4-42e7-a0a3-962b9b32060f)

pranavrajs commented on app/javascript/widget/views/ArticleViewer.vue:
```diff
@@ -1,16 +1,24 @@
 <template>
   <div class="bg-white h-full">
-    <IframeLoader :url="$route.query.link" />
+    <IframeLoader :url="$route.query.link" :is-rtl="isRTL" />
```
I guess we need to pass the dark mode as well in the URL.

iamsivin replied:
@pranavrajs Fixed https://github.com/chatwoot/chatwoot/pull/11022/commits/571a342dae4008613361b4860d013af14f913134

pranavrajs commented on app/javascript/widget/router.js:
```diff
@@ -8,12 +14,12 @@ const router = createRouter({
       path: '/unread-messages',
       name: 'unread-messages',
-      component: () => import('./views/UnreadMessages.vue'),
+      component: UnreadMessages,
```
Any reason why this was changed?

iamsivin replied:
@pranavrajs No particular reason, I took it from the branch you worked on [feat/widget-v2](https://github.com/chatwoot/chatwoot/pull/11016/files#diff-89ef130de6d1bd8967c9eb374df55d675040c3cefab5bdfc90a093688d13a111).
